noir-projects/aztec-nr/aztec/src/note/note_getter.nr

```diff
@@ -234,7 +245,7 @@ unconstrained pub fn view_notes<Note, N, M>(
         placeholder_opt_notes,
         placeholder_fields,
         placeholder_note_length
-    )
+    ).make_constrained(|opt_notes| opt_notes) // <- we're not constraining anything, but we don't want to pollute the API
```

Comment:
The only choices we have here (in an `unconstrained` public API function) are either:
 - to bubble up the `Unconstrained` value, polluting all callsites and forcing usage of `transform` (potentially unnecessarily if the value is ultimately not constrained)
 - to unwrap how I did here, losing knowledge of the fact that this value is unconstrained (which in this case is irrelevant because we never call this from constrained functions)

The issue is we currently cannot know if constrained functions will access this value at some point in their call stack, and can't decide which type to return.

Reply:
I know it's not ideal, but what about an `.unwrap()` function just like in Rust? I think there's value in this approach, and this annoyance can be solved by a function that basically says: "gimme the result, I know what I am doing" as opposed to `make_constrained` which would be used when you *actually* do something

Reply:
Would having an intrinsic `within_unconstrained() -> bool` available in the language (that gets resolved at compile time) help here?

Reply:
Ah, no, it wouldn't, since it's already inside an explicitely unconstrained fn

Reply:
> I know it's not ideal, but what about an .unwrap() function just like in Rust? I think there's value in this approach, and this annoyance can be solved by a function that basically says: "gimme the result, I know what I am doing" as opposed to make_constrained which would be used when you actually do something

I meant sth like this in my original comment. Yes it's not great, but maybe better than what we have now? (or maybe not worth it?)

The API taking constrainer will not work in general, I think: e.g., you could have an `assert(a - b == c)` which constrains 3 things at the same time.

Reply:
And what about mixing this approach (this is a prototype branch) with the wrapper?
https://github.com/AztecProtocol/aztec-packages/blob/c9e97b6826a8a586b58649c99a61613f7dda210b/noir/noir-repo/noir_stdlib/src/field/bn254.nr#L104
The explicitely unconstrained hints could return the wrapped versions

Reply:
I don't think that'd compile? Because you'd have a return type (wrapped or no) that depends on who called you.